tsconfig.check.json

```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "./tsconfig.json",
+  "include": ["src", "immutable", "infinite", "test/type"]
```

Comment:
I think we can just add `test/type` to tsconfig.json in rootDir. no need to have an extra config

Reply:
As mentioned in the description including `"test/type"` in tsconfig.json will also output the type of the test file in dist and get distributed to npm.

Reply:
I c, sounds good to merge!

Reply:
I tried locally again didn't see the output of type. did you run `tsc` without `--noEmit` so that the types are generated?
filed #1507 to merge them